spyder_kernels/utils/iofuncs.py

```diff
@@ -288,58 +289,101 @@ def save_dictionary(data, filename):
-                    data.pop(name)
-                elif isinstance(data[name], (list, dict)):
-                    # Saving arrays nested in lists or dictionaries
-                    if isinstance(data[name], list):
-                        iterator = enumerate(data[name])
-                    else:
-                        iterator = iter(list(data[name].items()))
-                    to_remove = []
-                    for index, value in iterator:
-                        if isinstance(value, np.ndarray) and value.size > 0:
-                            fname = __save_array(value, arr_fname,
-                                               len(saved_arrays))
-                            saved_arrays[(name, index)] = osp.basename(fname)
-                            to_remove.append(index)
-                    for index in sorted(to_remove, reverse=True):
-                        data[name].pop(index)
+                try:
+                    if isinstance(data[name],
+                                  np.ndarray) and data[name].size > 0:
+                        # Save arrays at data root
+                        fname = __save_array(data[name], arr_fname,
+                                             len(saved_arrays))
+                        saved_arrays[(name, None)] = osp.basename(fname)
+                        data.pop(name)
+                    elif isinstance(data[name], (list, dict)):
+                        # Save arrays nested in lists or dictionaries
+                        if isinstance(data[name], list):
+                            iterator = enumerate(data[name])
+                        else:
+                            iterator = iter(list(data[name].items()))
+                        to_remove = []
+                        for index, value in iterator:
+                            if isinstance(value,
+                                          np.ndarray) and value.size > 0:
+                                fname = __save_array(value, arr_fname,
+                                                     len(saved_arrays))
+                                saved_arrays[(name, index)] = (
+                                    osp.basename(fname))
+                                to_remove.append(index)
+                        for index in sorted(to_remove, reverse=True):
+                            data[name].pop(index)
+                    # If an array can't be saved with numpy for some reason,
+                    # leave the object intact and try to save it normally.
+                except (RuntimeError, pickle.PicklingError, TypeError):
+                    pass
             if saved_arrays:
                 data['__saved_arrays__'] = saved_arrays
-        pickle_filename = osp.splitext(filename)[0]+'.pickle'
+
+        pickle_filename = osp.splitext(filename)[0] + '.pickle'
+        # Attempt to pickle everything.
+        # If pickling fails, iterate through to eliminate problem objs & retry.
```

Comment:
`problem` -> `problematic`? (I don't know if that's the right word in this context or not).

Reply:
Either would work; "Problem <noun>" specifically refers to one or more items in a larger group of a specific type causing an issue, which seems appropriate here.